Trouble Compiling v0.10.6 to `wasm32-unknown-unknown`

```txt
error[E0599]: no method named `add` found for mutable reference `&mut [u8]` in the current scope
   --> /Users/nicksrandall/.cargo/registry/src/index.crates.io-6f17d22bba15001f/simd-json-0.10.6/src/simd128/deser.rs:117:31
    |
117 |                         input.add(idx),
    |                               ^^^ method not found in `&mut [u8]`
```

Compiling version `0.10.5` works fine.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -15,9 +15,9 @@ jobs:
           - windows-latest
           - macOS-latest
         rustflags:
-          - '-C target-cpu=native'
-          - '-C target-cpu=native -C target-feature=-avx2'
-          - '-C target-cpu=native -C target-feature=-avx2,-pclmulqdq'
+          - ''
+          - '-C target-feature=-avx2'
+          - '-C target-feature=-avx2,-pclmulqdq'
         features:
           - '--no-default-features'
           - ''
@@ -70,3 +70,27 @@ jobs:
           files: ./lcov.txt
           flags: unittests
           verbose: true
+  wasm:
+    strategy:
+      matrix:
+        features:
+          - '--no-default-features'
+          - ''
+          - '--features value-no-dup-keys'
+          - '--features known-key'
+          - '--features 128bit'
+          - '--features beef'
+          - '--features beef,known-key'
+          - '--features beef,known-key,128bit'
+          - '--features approx-number-parsing'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: dtolnay/rust-toolchain@stable
+      with:
+        components: llvm-tools-preview
+        targets: wasm32-unknown-unknown
+
+    - name: Compile
+      run: cargo build --target wasm32-unknown-unknown --features no-inline ${{ matrix.features }}

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -42,7 +42,7 @@ jobs:
     - name: Run tests (alloc)
       if: matrix.features == ''
       env:
-        RUSTFLAGS: ${{ matrix.rustflags }}
+        RUSTFLAGS: "-C target-cpu=native ${{ matrix.rustflags }}"
       run: cargo test --features alloc
 
     - name: Run tests

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,6 +14,9 @@ rust-version = "1.61"
 [package.metadata.docs.rs]
 features = ["docsrs"]
 
+[target.'cfg(target_family = "wasm")'.dependencies]
+getrandom = { version = "0.2", features = ["js"] }
+
 [dependencies]
 lexical-core = { version = "0.8", features = ["format"] }
 beef = { version = "0.5", optional = true }
@@ -100,6 +103,9 @@ hints = [] # requires nightly
 
 # for perf testing, used by the example
 perf = ["perfcnt", "getopts", "colored", "serde_json"]
+
+# for documentation
+docsrs = []
 
 [[example]]
 name = "perf"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1677,6 +1677,14 @@ mod tests_serde {
     }
 
     #[test]
+    fn invalid_float_array() {
+        let mut data =
+            b"[11111111111111111111111111111E1,-111111111111111111111E111111111".to_vec();
+
+        assert!(to_owned_value(&mut data).is_err());
+    }
+
+    #[test]
     fn crazy_string() {
         // there is unicode in here!
         let d = "\"𐀀𐀀  𐀀𐀀0 𐀀A\\u00000A0 A \\u000b\"";

--- a/src/numberparse.rs
+++ b/src/numberparse.rs
@@ -109,7 +109,10 @@ fn parse_eight_digits_unrolled(chars: &[u8]) -> u32 {
 }
 
 #[cfg_attr(not(feature = "no-inline"), inline)]
-#[cfg(any(target_feature = "neon", target_feature = "simd128"))]
+#[cfg(all(
+    any(target_feature = "neon", target_feature = "simd128"),
+    feature = "swar-number-parsing"
+))]
 #[allow(clippy::cast_ptr_alignment)]
 fn parse_eight_digits_unrolled(chars: &[u8]) -> u32 {
     let val = unsafe { (chars.as_ptr() as *const u64).read_unaligned() }; //    memcpy(&val, chars, sizeof(u64));

--- a/src/numberparse/correct.rs
+++ b/src/numberparse/correct.rs
@@ -364,7 +364,7 @@ fn f64_from_parts_slow(slice: &[u8], offset: usize) -> Result<StaticNode> {
     ) {
         Ok(val) => {
             if val.is_infinite() {
-                err!(offset, get!(slice, offset.saturating_sub(1)))
+                err!(offset, get!(slice, 0))
             }
 
             Ok(StaticNode::F64(val))

--- a/src/simd128/deser.rs
+++ b/src/simd128/deser.rs
@@ -26,7 +26,6 @@ impl<'de> Deserializer<'de> {
         mut idx: usize,
     ) -> Result<&'de str> {
         use ErrorType::{InvalidEscape, InvalidUnicodeCodepoint};
-        let input: &mut [u8] = unsafe { std::mem::transmute(input) };
         // Add 1 to skip the initial "
         idx += 1;
 

